src/GameLogic/Entities/Protagonist.ts

```diff
@@ -0,0 +1,32 @@
+import {Entity} from "../Entity";
+import {EntityType, PlayerAction, PlayerActionUtils} from "../Enums";
+
+// @todo I think we should only have a single class for the player entity, to ensure we never treat them differently based on how they move
+export class Protagonist implements Entity {
+	public readonly type: EntityType;
+	public x: number;
+	public y: number;
+
+	public isPlayerControlled: boolean;
+	public movesQueue: PlayerAction[];
+
+	constructor(isPlayerControlled: boolean = true) {
+		this.type = EntityType.Protagonist;
+		this.x = 0;
+		this.y = 0;
+		this.isPlayerControlled = isPlayerControlled;
+		this.movesQueue = [];
+	}
+
+	public update(): void {
+		const action = this.movesQueue.pop();
+
+		if (action === undefined) {
+			return;
```

Comment:
Might be worth putting a todo here to fill in what we want to happen if a timeclone makes it to the end of their timeline while still alive (like you changed history so they don't die). On a related note, that's a thing we need to work out in the design at some point.

Reply:
👍